### PR TITLE
Update scripts in sonic-buildimage from py-swsssdk to swsscommon

### PR DESCRIPTION
**Pull request by liuh-80**
#### Why I did it
Update scripts in sonic-buildimage from py-swsssdk to swsscommon

#### How I did it
Change code to use swsscommon.

#### How to verify it
Pass all E2E test case

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

#### Description for the changelog
Update scripts in sonic-buildimage from py-swsssdk to swsscommon

### CODE REVIEW

**Comment by liuh-80** [comment]
Following code in bgpmon.py seems has different behavior with py-swsssdk, will investigate and fix:

        self.db = swsscommon.SonicV2Connector()
        self.db.connect(self.db.STATE_DB, False)
        self.client = self.db.get_redis_client(self.db.STATE_DB)

........

        for key, value in data.items():
            if value is None:
                # delete case
                self.client.del(key)
            else:
                # Add or Modify case
                self.client.hmset(key, value)

File: src/sonic-bgpcfgd/bgpmon/bgpmon.py [hunk 1]
```diff
@@ -43,10 +43,9 @@ def __init__(self):
         self.new_peer_l = set()
         self.new_peer_state = {}
         self.cached_timestamp = 0
-        self.db = swsssdk.SonicV2Connector()
+        self.db = swsscommon.SonicV2Connector()
         self.db.connect(self.db.STATE_DB, False)
-        client = self.db.get_redis_client(self.db.STATE_DB)
-        self.pipe = client.pipeline()
+        self.pipe = self.db.get_redis_client(self.db.STATE_DB)
```
Comment (liuh-80) [inline src/sonic-bgpcfgd/bgpmon/bgpmon.py:48]: Remove pipe because in flush_pipe(self, data) has following code:
            if value is None:
                # delete case
                self.pipe.delete(key)
            else:
                # Add or Modify case
                self.db.hmset(self.pipe.STATE_DB. key, value)


This is because swsscommon not using pyredis, and self.db.get_redis_client(self.db.STATE_DB) will return StrictRedis. so the StrictRedis.pipeline() method does not exist in sonic-swss-common.

Also another reason is the DBConnector only has following 1 method in swsscommon.py:

    def hmset(self, multiHash):
        return _swsscommon.DBConnector_hmset(self, multiHash)

**Comment by liuh-80** [comment]
> This pull request **introduces 1 alert** and **fixes 3** when merging [d557724](https://github.com/Azure/sonic-buildimage/commit/d55772417b857779ec17f320742c1d687020c5fd) into [1568667](https://github.com/Azure/sonic-buildimage/commit/15686670116bf4ca01842a3e6f681a751e716bb5) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-buildimage/rev/pr-694f158b266c562e43a8a924f86f5a0dad767152)
> 
> **new alerts:**
> 
> * 1 for Module is imported more than once
> 
> **fixed alerts:**
> 
> * 3 for Unused import

Fixed.

File: src/sonic-py-common/sonic_py_common/util.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+from __future__ import print_function
+import json
+import logging
+import logging.config
+import sys
+from getopt import getopt
+
+
+# TODO: move to dbsync project.
```
Comment (liuh-80) [inline src/sonic-py-common/sonic_py_common/util.py:13]: Can't move to dbsync, because currently snmp agent also using this:
https://github.com/Azure/sonic-snmpagent/blob/5f8f475b54fbe12b959258bb5567f71577aff0ea/src/sonic_ax_impl/__main__.py

**Comment by qiluo-msft** [comment]
> # led_control.py

We can move all the changes in platform/ and device/ paths into another PR. Those changes need to be tested with proper hardware, reviewed and approved by platform vendors.

Other platform-independent changes are relative easy to test.

---
Refers to: device/accton/x86_64-accton_minipack-r0/plugins/led_control.py:1 in 1bcf1fd. [](commit_id = 1bcf1fd7cd9aa83264e4c99182c0dc5287ce2f3d, deletion_comment = False)

File: src/sonic-bgpcfgd/bgpmon/bgpmon.py [hunk 3]
```diff
@@ -106,11 +104,10 @@ def flush_pipe(self, data):
         for key, value in data.items():
             if value is None:
                 # delete case
-                self.pipe.delete(key)
+                self.db.delete(self.db.STATE_DB, key)
```
Comment (qiluo-msft) [inline src/sonic-bgpcfgd/bgpmon/bgpmon.py:107]: >db [](http://example.com/codeflow?start=21&length=2)

Pipeline is more efficient on a batch of similar operations.

File: src/sonic-py-common/sonic_py_common/port_util.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
+"""
+Bridge/Port mapping utility library.
```
Comment (qiluo-msft) [inline src/sonic-py-common/sonic_py_common/port_util.py:2]: If these files are simply moved from another repo, let's separate them into a standalone PR, not to mix with other changes not of the same nature.

File: src/sonic-bgpcfgd/bgpmon/bgpmon.py (hunk 3 above)
Comment (qiluo-msft) [reply src/sonic-bgpcfgd/bgpmon/bgpmon.py:107]: Not an exact replacement of redis-py pipe, but you may check if `RedisPipeline` satisfy the usage here.

**Comment by liuh-80** [comment]
> > # led_control.py
> 
> We can move all the changes in platform/ and device/ paths into another PR. Those changes need to be tested with proper hardware, reviewed and approved by platform vendors.
> 
> Other platform-independent changes are relative easy to test.
> 
> Refers to: device/accton/x86_64-accton_minipack-r0/plugins/led_control.py:1 in [1bcf1fd](https://github.com/Azure/sonic-buildimage/commit/1bcf1fd7cd9aa83264e4c99182c0dc5287ce2f3d). [](commit_id = [1bcf1fd](https://github.com/Azure/sonic-buildimage/commit/1bcf1fd7cd9aa83264e4c99182c0dc5287ce2f3d), deletion_comment = False)

All hardware related change move to following PRs for different vender:
[[Accton] Remove unused swsssdk import from accton device code by liuh-80 · Pull Request #11348 · Azure/sonic-buildimage (github.com)](https://github.com/Azure/sonic-buildimage/pull/11348)
[[AlphaNetworks] Replace swsssdk with swsscommon in alphanetworks devices by liuh-80 · Pull Request #11349 · Azure/sonic-buildimage (github.com)](https://github.com/Azure/sonic-buildimage/pull/11349)
[[Centec] Replace swsssdk with swsscommon in centec devices. by liuh-80 · Pull Request #11350 · Azure/sonic-buildimage (github.com)](https://github.com/Azure/sonic-buildimage/pull/11350)

File: src/sonic-bgpcfgd/bgpmon/bgpmon.py (hunk 3 above)
Comment (liuh-80) [reply src/sonic-bgpcfgd/bgpmon/bgpmon.py:107]: The RedisPipeline  not have delete and hmset methods, to use RedisPipeline we need build a new RedisCommand command and run the command with RedisPipeline. that basically rewrite the delete and hmset method in DBConnector:

https://github.com/Azure/sonic-swss-common/blob/15c0f729648ee9f26ac9aea767d5712db34ca298/common/dbconnector.cpp#L862

File: dockers/docker-sflow/port_index_mapper.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 import traceback
 from sonic_py_common.logger import Logger
 from socket import if_nametoindex
-from swsssdk import SonicV2Connector, port_util
+from sonic_py_common import port_util
```
Comment (liuh-80) [inline dockers/docker-sflow/port_index_mapper.py:8]: The port_util change split to another PR, this PR need wait for another PR merge first:
https://github.com/Azure/sonic-buildimage/pull/11347

File: src/sonic-bgpcfgd/bgpmon/bgpmon.py (hunk 3 above)
Comment (qiluo-msft) [reply src/sonic-bgpcfgd/bgpmon/bgpmon.py:107]: `delete` should be straightforward. We need more discussion on `hmset`

Comment (liuh-80) [reply src/sonic-bgpcfgd/bgpmon/bgpmon.py:107]: I will check if we can improve current hmset without using lua script.

Comment (liuh-80) [reply src/sonic-bgpcfgd/bgpmon/bgpmon.py:107]: I found the hmset here actually using following API which is not lua script based:

template<typename InputIterator>
void DBConnector::hmset(const std::string &key, InputIterator start, InputIterator stop)
{
    RedisCommand shmset;
    shmset.formatHMSET(key, start, stop);
    RedisReply r(this, shmset, REDIS_REPLY_STATUS);
}

So maybe we can finish this PR first, and improve the lua based hmset with another PR.

Comment (qiluo-msft) [reply src/sonic-bgpcfgd/bgpmon/bgpmon.py:107]: Could you postpone this change to another future PR? we can merge the other files in this PR first.

Comment (liuh-80) [reply src/sonic-bgpcfgd/bgpmon/bgpmon.py]: Fixed, change in this file reverted.

**Review by qiluo-msft — APPROVED** [review]
